src/frontend/src/utils/column_index_mapping.rs

```diff
@@ -340,6 +343,40 @@ impl ColIndexMapping {
         }
     }
 
+    /// Rewrite the indices in a functional dependency.
+    ///
+    /// If some columns in the `from` side are removed, then this fd is no longer valid. For
+    /// example, for ABC --> D, it means that A, B, and C together can determine C. But if B is
+    /// removed, this fd is not valid. For this case, we will return [`None`]
+    ///
+    /// Additionally, If the `to` side of a functional dependency becomes empty after rewriting, it
+    /// means that this dependency is unneeded so we also return [`None`].
```

Comment:
Looks good! But I'm wondering another problem: will we need to simplify a FD set before rewriting? 🤔

Reply:
e.g., `minimize_key`, or even compute all closures for all combination of columns..?

Reply:
But even current behavior is not precise, it is conservative (or correct), so OK to me

Reply:
example:
FD set: {AB->C, A->B}, rewrite: remove B.
current: {}
ideally: {A->C}

Reply:
I think there is no need to keep empty `fd`s . So maybe simplification is OK here?

Reply:
> example:
FD set: {AB->C, A->B}, rewrite: remove B.
current: {}
ideally: {A->C}

Good catch. Let me reconsider this.

Reply:
Just add some comments about this in efb3d59164d3a36baec2c319993febfe7fa9dd9e